opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -2184,8 +2222,10 @@ namespace Opm
             // calculating the perforation rate for each perforation that belongs to this segment
             const Scalar seg_pressure = getValue(this->primary_variables_.getSegmentPressure(seg));
             for (const int perf : this->segments_.perforations()[seg]) {
-                const int local_perf_index = this->pw_info_.globalToLocal(perf);
-                if (local_perf_index < 0) // then the perforation is not on this process
+                const int local_perf_index = this->pw_info_.activeToLocal(perf);
+                // If local_perf_index == -1, then the perforation is not on this process
+                // If this->well_cells_.size() == 0, then this process contains only shut perforations
+                if (local_perf_index < 0 or this->well_cells_.size() == 0)
```

Comment:
Are you sure about the right hand side of the `or`? It seems to change the semantics for undistributed wells, but maybe the rhs is never true in that case.

Maybe we are computing values for shut perforations for a reason? We seem to do that for undistributed wells before this PR it seems.